snuba/clusters/cluster.py

```diff
@@ -1,15 +1,55 @@
 from abc import ABC, abstractmethod
-from typing import Any, Callable, Generic, Mapping, Optional, Set, TypeVar
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Mapping,
+    MutableMapping,
+    Optional,
+    Set,
+    Tuple,
+    TypeVar,
+)
 
 from snuba import settings
 from snuba.clickhouse.http import HTTPBatchWriter
 from snuba.clickhouse.native import ClickhousePool, NativeDriverReader
 from snuba.clickhouse.sql import SqlQuery
 from snuba.clusters.storage_sets import StorageSetKey
+from snuba.optimize import OPTIMIZE_SEND_RECEIVE_TIMEOUT
 from snuba.reader import Reader, TQuery
 from snuba.writer import BatchWriter, WriterTableRow
 
 
+ClickhouseClientSettingsType = Tuple[Mapping[str, Any], Optional[int]]
```

Comment:
This seems a good candidate for a NamedTuple so people know what that Optional[int] is.